plugins/flytekit-memray/README.md

```diff
@@ -0,0 +1,55 @@
+# Memray Profiling Plugin
+
+Memray tracks and reports memory allocations, both in python code and in compiled extension modules.
+This Memray Profiling plugin enables memory tracking on the Flyte task level and renders a memgraph profiling graph on Flyte Deck.
+
+To install the plugin, run the following command:
+
+```bash
+pip install flytekitplugins-memray
+```
+
+Example
+```python
+from flytekit import workflow, task, ImageSpec
+from flytekitplugins.memray import memray_profiling
+import time
+
+
+image = ImageSpec(
+    name="memray_demo",
+    packages=["flytekitplugins_memray"],
+    env={"PYTHONMALLOC": "malloc"},
```

Comment:
Instead of hard coding this into the environment, can we now `trace_python_allocators=True`?

Reply:
I tested that and its not throwing any warnings but the results look different though:
Thats the task I tested without having the env variable set:
``` python
@task(container_image=image, enable_deck=True)
@memray_profiling(trace_python_allocators=True, memray_reporter_args=["--leaks"])
def memory_leakage(n: int) -> str:
    generate_data(n=n)

    return "Well"
```

Thats the result:
![image](https://github.com/user-attachments/assets/7d43b6d0-688e-4c63-91ac-861c1c972fbd)

Not sure if this is expected

Reply:
For completeness, what do you see when you set `trace_python_allocators=False`?

Reply:
This, which is expected i guess
![image](https://github.com/user-attachments/assets/26afddb7-9869-40ef-94f9-3d3b304a366d)

Reply:
It's weird that it gives two different flamegraphs. The new flamegraph makes more sense to me because the tracker is wrapping the user code and you can clearly see the `generete_data`.

I can not really see where the `generate_data` is on your original flamegraph.

Reply:
Oh man, I mixed up my demos. Its looking exactly the same with `trace_python_allocaters=False` and having the env variable set. Sorry for the confusion, I will update the Readme shortly
![image](https://github.com/user-attachments/assets/a02bdfb3-dede-4b20-a124-3194fa0d25ca)